### PR TITLE
New settings option to hide menus from widget

### PR DESCRIPTION
**Pull request by bucko13**
Thought it would be fun to experiment with "display settings" that interact with other widgets. Would be similar to what we want for an advanced mode in bwallet. Unfortunately, persistence won't work because of a [bug in bpanel](https://github.com/bpanel-org/bpanel/issues/171), but it's an interesting proof of concept.

### CODE REVIEW

**Comment by pinheadmz** [comment]
Looks cool so far! Great idea. Only funny thing is how the button changes shape!
untitled mov

**Comment by bucko13** [comment]
Agreed. We need to change that in bpanel-ui though. Just a styling issue which is probably coming up in other places too.

File: lib/PriceSettings.js [hunk 1]
```diff
@@ -33,54 +37,80 @@ class PriceSettings extends Component {
...
+              {formattedChain || '...'}
+            </span>
+            ).
+          </Text>
+          <Text>
+            {price.crypto +
+              ' ' +
+              price.fiatSymbol +
+              this.moneyNumber(price.price) +
+              ' '}
```
Comment (pinheadmz) [inline lib/PriceSettings.js]: Dunno why this space doesn't look right in settings, it looks good in the footer?
I tried replacing this with a `&nbsp;` and got that little margin back between the elements

File: lib/PriceSettings.js [hunk 2]
```diff
@@ -33,54 +37,80 @@ class PriceSettings extends Component {
...
+              <option value={val} key={val}>
+                {val}
+              </option>
+            ))}
+          </select>
+        </div>
+        <div className="row">
+          <Header type="h5">Settings for Widget Display</Header>
+          <Label
+            text="Show settings menu?"
```
Comment (pinheadmz) [inline lib/PriceSettings.js]: Kinda think you can lose this question -- there's triple redundancy explaining that button right there!

screen shot 2019-01-14 at 11 42 49 pm

File: lib/PriceSettings.js (hunk 1 above)
Comment (bucko13) [reply lib/PriceSettings.js]: Stupid JSX

**Comment by bucko13** [comment]
Updated:
screen shot 2019-01-15 at 10 42 09 am

**Comment by pinheadmz** [comment]
Yeah looks dope! Wanna push so I can play locally?

**Comment by bucko13** [comment]
Oh yeah, whoops! One sec

File: lib/PriceSettings.js [hunk 3]
```diff
@@ -33,54 +37,76 @@ class PriceSettings extends Component {
...
+            ))}
+          </select>
+        </div>
+        <div className="row">
+          <Header type="h5">Settings for Widget Display</Header>
+          <Label text="Show/Hide dropdown menus?" stacked={false}>
+            {hideMenus ? (
+              <Button onClick={() => hideWidgetMenus(false)}>Show Menus</Button>
+            ) : (
+              <Button type="action" onClick={() => hideWidgetMenus(true)}>
```
Comment (pinheadmz) [inline lib/PriceSettings.js:104]: ` type="action"` you only need this on HIDE and not SHOW? Is that what makes the button change colors when its switched?

Comment (bucko13) [reply lib/PriceSettings.js:104]: yep! We only have two button types at the moment, default and `action`, and I wanted it to change so it was more obvious. Would be nice to add more in the future.

**Comment by pinheadmz** [comment]
Cool! digging this - we just need to make those two button types the same shape!
`//TODO`

**Review by pinheadmz — APPROVED** [review]

**Comment by bucko13** [comment]
Added issue to bpanel-ui: https://github.com/bpanel-org/bpanel-ui/issues/49

**Pull request merged**